src/ServiceControl.Monitoring/Infrastructure/EndpointInstanceActivityTracker.cs

```diff
@@ -0,0 +1,28 @@
+﻿namespace ServiceControl.Monitoring.Infrastructure
+{
+    using System;
+    using System.Collections.Concurrent;
+
+    public class EndpointInstanceActivityTracker
+    {
+        ConcurrentDictionary<EndpointInstanceId, DateTime> endpointsInstances = new ConcurrentDictionary<EndpointInstanceId, DateTime>();
+        public static readonly TimeSpan StalenessThreshold = TimeSpan.FromMinutes(5);
```

Comment:
We must just load this time from the Settings class. We'd have to introduce a new setting for it too.